subprojects/architecture-test/src/test/java/org/gradle/architecture/test/PublicApiCorrectnessTest.java

```diff
@@ -70,7 +71,11 @@ public class PublicApiCorrectnessTest {
                 .or(type(BigDecimal.class))
                 .or(type(Element.class))
                 .or(type(QName.class))
-                .as("built-in JDK classes"))
+                .and(
+                    DescribedPredicate.not(resideInAnyPackage("java.util.function"))
+                        .or(type(BiFunction.class))
+                )
```

Comment:
❌ Let's not do this, but remove `java.util.function` from line 66. And add some comment that we don't want to have `java.util.function` including the reasons why.

Reply:
Thanks, missed that it's already ther